packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -56,7 +55,8 @@ export class IndexedDbQueryCache implements QueryCache {
   private metadata = new DbTargetGlobal(
     /*highestTargetId=*/ 0,
     /*lastListenSequenceNumber=*/ 0,
-    SnapshotVersion.MIN.toTimestamp()
+    SnapshotVersion.MIN.toTimestamp(),
+    /*targetCount=*/ 0
   );
```

Comment:
Since you've reworked this so that the DBTargetGlobal always exists, I think we don't need to initialize metadata here, and I'd prefer not have these defaults duplicated unnecessarily between here and indexeddb_schema.ts.  So perhaps change this to just:

private metadata: DbTargetGlobal = null;

Reply:
Done, and fixed the bug in the tests where we weren't calling `start()` but didn't know because we were initializing this to a dummy value.

So, good suggestion!